modules/applications/pages/ui-builder/how-to-upload-multiple-documents.adoc

```diff
@@ -0,0 +1,91 @@
+= Upload multiple documents simultaneously
+:page-aliases: applications:how-to-upload-multiple-documents.adoc
+:description: Learn how to upload multiple documents simultaneously
+
+{description}
+
+* Difficulty: intermediate
+* Prerequisites: have an xref:create-an-interface.adoc[interface created] on Bonita UI Builder
+
+If you need to upload multiple files using Bonita's formFileUpload servlet, you'll need to process each file separately by looping through them, adding a parameter to the API query for each file. Since Bonita's `formFileUpload` doesn't support uploading multiple files at once, you'll also need to synchronize the uploads.
+
+
+== Example
+
+Suppose you want to upload multiple files at the same time using a `FilePicker` widget in the Bonita UI Builder.
+
+=== 1. Use a FilePicker widget
+To do so, drag and drop a `FilePicker` widget onto your interface and configure it as follows:
+
+image:ui-builder/guides/file-picker-widget-properties.png[file-picker-widget-propeties]
+
+[TIP]
+In this example, we don't set a limit for the `Max no. of files` because we're allowing multiple files to be uploaded. However, you can set a limit on the number of files to be uploaded, as well as adjust other properties if needed.
+
+
+=== 2. Create a post request
+* Go to `Queries` section and create a new query/API and configure it as follows:
+
+image:ui-builder/guides/upload-files-query.png[upload-files-query]
+
+
+=== 3. Create JavaScript object
+
+* Go to `JS` section and create a new JS object to create functions given below that allows to upload multiple files simultaneously.
+
+* Function to get files
+[source, JS]
+----
+export default {
+     async getFileUpload () {
+		const files = testFileUpload.files;
+
+		const promises = files.map((file) => {
+			return upload_files.run({ fileName: file });
+		});
+
+		try {
+			const results = await Promise.all(promises);
+			const santizedResult = results.map(result => JSON.parse(result));
+			return santizedResult;
+		} catch (error) {
+			console.error('The error is ', error);
+			throw error;
+		}
+	}
+ }
+----
+
+The `getFileUpload` function is an asynchronous function that uploads multiple files concurrently. It retrieves the files from `testFileUpload.files` and maps each file to an upload promise using `upload_files.run()`.
+These promises are then executed simultaneously using `Promise.all`. Once all uploads are complete, the function parses the results into JSON objects and returns them.
+The function also includes error handling, logging any errors that occur during the upload process and re-throwing them for further handling.
+
+* Function to upload files
+
+[source, JS]
+----
+export default {
+     async uploadFiles() {
+		if(testFileUpload.files.length > 2) {
```

Comment:
why this `> 2` and not  `>1` or `>=2` ?

Reply:
Because we need to check if at least 3 files are provided, as required. It can be >=3, but not >=2 which doesn't match the requirement.